mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -562,7 +562,9 @@ public void runVoid() throws MailboxException {
                     }
                 }, true);
 
-            return Optional.fromNullable(Iterables.getLast(mailboxIds));
+            if (!mailboxIds.isEmpty()) {
```

Comment:
Nope, the code might create "hierarchical" mailboxes (aka: a.b.c.d), and needs to create first a.b, a.b.c, a.b.c.d, thus we can not "simply" return the mailbox Id.